src/routes/auth/utils.js

```diff
@@ -0,0 +1,18 @@
+import jwt      from 'jsonwebtoken';
+import url      from 'url';
+
+import config   from '../../config';
+
+export function finalizeAuth(req, res) {
+  const token = jwt.sign(req.user, config.get('adminSessionSecret'));
+
+  if (req.session && req.session.redirectUrl) {
+    const redirectUrl = url.parse(req.session.redirectUrl, true);
+    redirectUrl.query.token = token;
+    req.session.destroy();
```

Comment:
I think I'll need to destroy this in case of failures too... but not so easy to do with the current way of managing errors :)